src/rgw/motr/gc/gc.cc

```diff
@@ -47,21 +47,42 @@ void *MotrGC::GCWorker::entry() {
         << worker_id << " Working on GC Queue: " << iname << dendl;
       // time based while loop
       do {
-        // fetch the next entry from index "iname"
-        // check if the object is ready for deletion
-        // for the motr object
+        bufferlist bl;
+        std::vector<std::string>keys(motr_gc->max_count + 1);
+        std::vector<bufferlist>vals(motr_gc->max_count + 1);
+        uint32_t rgw_gc_obj_min_wait = cct->_conf->rgw_gc_obj_min_wait;
+        rc = motr_gc->store->next_query_by_name(iname,
+                                    keys, vals, obj_exp_time_prefix);
+        if (rc < 0) {
+          // In case of failure, worker will keep retrying till end_time
+          ldpp_dout(dpp, 0) <<__func__<<": ERROR: NEXT query failed. rc=" << rc << dendl;
+        }
 
-        // delete that object
-        // rc = dequeue();
-
-        // remove the entry from iname
-
-        // Exit the loop if required work is complete
-        processed_count++;
-        if (processed_count >= motr_gc->max_count) break;
+        // fetch entries as per defined in rgw_gc_max_trim_chunk from index iname
+        for (uint32_t j = 0; j < motr_gc->max_count && !keys[j].empty(); j++) {
+          bufferlist::const_iterator blitr = vals[j].cbegin();
+          motr_gc_obj_info ginfo;
+          ginfo.decode(blitr);
+          // check if the object is ready for deletion
+          if(ginfo.deletion_time + rgw_gc_obj_min_wait < std::time(nullptr)) {
```

Comment:
since the entries are sorted by the time, if this condition is unsatisfied, we should break out of for and while loop.

Reply:
Put deletion in `else` block, `if` block will breakout if object is not expired.